src/Symfony/Component/DomCrawler/Form.php

```diff
@@ -454,22 +454,49 @@ private function addField(\DOMElement $node)
         }
 
         $nodeName = $node->nodeName;
+        $label = $this->findAssociatedLabel($node);
         if ('select' == $nodeName || 'input' == $nodeName && 'checkbox' == strtolower($node->getAttribute('type'))) {
-            $this->set(new Field\ChoiceFormField($node));
+            $this->set(new Field\ChoiceFormField($node, $label));
         } elseif ('input' == $nodeName && 'radio' == strtolower($node->getAttribute('type'))) {
             // there may be other fields with the same name that are no choice
             // fields already registered (see https://github.com/symfony/symfony/issues/11689)
             if ($this->has($node->getAttribute('name')) && $this->get($node->getAttribute('name')) instanceof ChoiceFormField) {
                 $this->get($node->getAttribute('name'))->addChoice($node);
             } else {
-                $this->set(new Field\ChoiceFormField($node));
+                $this->set(new Field\ChoiceFormField($node, $label));
             }
         } elseif ('input' == $nodeName && 'file' == strtolower($node->getAttribute('type'))) {
-            $this->set(new Field\FileFormField($node));
+            $this->set(new Field\FileFormField($node, $label));
         } elseif ('input' == $nodeName && !in_array(strtolower($node->getAttribute('type')), array('submit', 'button', 'image'))) {
-            $this->set(new Field\InputFormField($node));
+            $this->set(new Field\InputFormField($node, $label));
         } elseif ('textarea' == $nodeName) {
-            $this->set(new Field\TextareaFormField($node));
+            $this->set(new Field\TextareaFormField($node, $label));
         }
     }
+
+    /**
+     * Finds the associated label to a given field node.
+     *
+     * @param \DOMElement $node The \DOMElement instance of the field
+     *
+     * @return \DOMNode|null The \DOMElement of the associated label tag or null if none was found
+     */
+    private function findAssociatedLabel(\DOMElement $node)
```

Comment:
Have you considered moving this implementation to the FormField's `getLabel()` method? You wouldn't have to add a new argument to FormField's constructor, and the logic wouldn't be called if the label wasn't accessed.  `ownerDocument` property is available on the `$node` that's already passed to the FormField.